### DISCUSSION TITLE
connection pool support

### DISCUSSION THREAD

**Discussion by cufewxy** [sfu-db/connector-x#205] [General]
Hi, will cx suuport connection pool?

**Comment by wangxiaoying** [comment]
Hi @cufewxy , thanks for open this discussion. We do not plan for this yet. connectorx targets more on the bulk loading scenario where the bottleneck is caused by the data size. So the connection construction overhead in this case should be negligible. But happy to see whether this is a common need!

**Reply by cufewxy** [reply]
Thanks, I get your point. It is a pity that our system must use connection pool, or database department will be crazy...If it is a common need in the future, maybe you can use create_engine like pandas, will decrease cost of changing code. engine itself contains connection pool

**Comment by gusutabopb** [comment]
It'd really be nice to have some control over connections (pooling/reusing).

Currently I have to access a MSSQL DB where establishing a connection takes more than 1s. When multiple queries have to be made in succession, having to open a new connection every time is a significant overhead.

**Reply by ignacevau** [reply]
This would be really useful, sometimes neither the data size or the connection time are "the bottleneck" because both times are being added up. Having a persistent connection could reduce the time for sequential queries quite a bit.
